proxy/upgrade.go

```diff
@@ -124,6 +124,11 @@ func (p *upgradeProxy) serveHTTP(w http.ResponseWriter, req *http.Request) {
 
 	if resp.StatusCode != http.StatusSwitchingProtocols {
 		log.Debugf("Got invalid status code from backend: %d", resp.StatusCode)
+		for key, values := range resp.Header {
+			for _, value := range values {
+				w.Header().Add(key, value)
+			}
+		}
```

Comment:
Why not https://pkg.go.dev/maps#Copy?

Also, this and below resembles why proxy.go is doing so I am wondering whether its worth it to extract the "serve backend response" logic and reuse it here.